File: python/posted/read.py
from pathlib import Path

import pandas as pd
import yaml


def read_csv_file(fpath: str):
    """
    Read CSV data file

    Parameters
    ----------
    fpath: str
        Path of the file to read
    Returns
    -------
        pd.DataFrame
            DataFrame containg the data of the CSV
    """
    return pd.read_csv(fpath)


def read_yml_file(fpath: Path):
    """
    Read YAML config file

    Parameters
    ----------
    fpath: str
        Path of the file to read
    Returns
    -------
        dict
            Dictionary containing config info
    """
    fhandle = open(fpath, 'r', encoding='utf-8')
    ret = yaml.load(stream=fhandle, Loader=yaml.FullLoader)
    fhandle.close()
    return ret

File: R/posted/read.R
# read CSV data file
read_csv_file <- function (fpath) {
    return(read.csv(
        fpath,
        sep=',',
        quote='"',
        encoding='utf-8'
    ))
}

# read YAML config file
read_yml_file <- function (fpath) {
    fhandle <- file(fpath, "r", encoding="utf-8")
    ret <- yaml::read_yaml(file=fhandle, fileEncoding="utf-8")
    close(fhandle)
    return(ret)
}